src/gocam/indexing/Indexer.py

```diff
@@ -1,23 +1,67 @@
 import logging
+from collections.abc import Iterable
 from functools import cached_property, lru_cache
-from typing import Optional, List, Collection, Tuple, Set
+from typing import Collection, List, Optional, Tuple
 
+import networkx as nx
+import pystow
+import yaml
 from oaklib import get_adapter
-from oaklib.datamodels.vocabulary import PART_OF, IS_A
+from oaklib.datamodels.vocabulary import IS_A, PART_OF
 
 from gocam.datamodel import (
+    Activity,
+    Association,
     EnabledByGeneProductAssociation,
     EnabledByProteinComplexAssociation,
     Model,
+    Object,
     PublicationObject,
-    TermAssociation,
+    QueryIndex,
     TermObject,
 )
-from gocam.datamodel import QueryIndex
-import networkx as nx
 
 logger = logging.getLogger(__name__)
 
+_PYSTOW_MODULE = pystow.module("gocam")
+_CURRENT_GROUPS_YAML_URL = "https://current.geneontology.org/metadata/groups.yaml"
+
+
+def _iter_association_references(
+    association: Association | None,
+) -> Iterable[str]:
+    """
+    Extract references from an Association.
```

Comment:
I'm partial to doctests -- but also partial to the argument that in this case the size of the doctest may actually occlude the already simple logic...